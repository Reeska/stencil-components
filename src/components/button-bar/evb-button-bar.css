:host {
  align-items: center;
  display: flex;
  margin: 1rem;
}
:host(.justify-left) {
  justify-content: flex-start;
}
:host(.justify-center) {
  justify-content: center;
}
:host(.justify-right) {
  justify-content: flex-end;
}

::slotted(evb-button + evb-button) {
  margin-left: 8px;
}
