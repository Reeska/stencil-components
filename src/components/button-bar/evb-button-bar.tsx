import { Component, Prop } from "@stencil/core";

/**
 * Container for buttons
 */
@Component({
  tag: 'evb-button-bar',
  styleUrl: 'evb-button-bar.css',
  shadow: false,
  scoped: true
})
export class EvbButtonBar {

  @Prop() justify: 'left' | 'center' | 'right';

  hostData() {
    return {
      class: `justify-${this.justify}`
    }
  }

  render() {
    return (
      <slot/>
    );
  }
}
