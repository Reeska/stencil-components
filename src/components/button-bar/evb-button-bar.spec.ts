import { EvbButtonBar } from './evb-button-bar';

describe('EvbButtonBar', () => {
  let component: EvbButtonBar;

  beforeEach(() => component = new EvbButtonBar());

  it('should render', () => {
    expect(component.render()).toMatchSnapshot();
  });
});
