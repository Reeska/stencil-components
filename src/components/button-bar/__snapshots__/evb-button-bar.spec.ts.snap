// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EvbButtonBar should add a justify class 1`] = `
Object {
  "$attrs$": Object {
    "class": "justify-center align-center",
  },
  "$children$": Array [
    Object {
      "$attrs$": null,
      "$children$": null,
      "$elm$": undefined,
      "$flags$": 0,
      "$key$": undefined,
      "$name$": undefined,
      "$tag$": "slot",
    },
  ],
  "$elm$": undefined,
  "$flags$": 0,
  "$key$": undefined,
  "$name$": undefined,
  "$tag$": Object {},
}
`;

exports[`EvbButtonBar should render 1`] = `
Object {
  "$attrs$": Object {
    "class": "justify-left align-center",
  },
  "$children$": Array [
    Object {
      "$attrs$": null,
      "$children$": null,
      "$elm$": undefined,
      "$flags$": 0,
      "$key$": undefined,
      "$name$": undefined,
      "$tag$": "slot",
    },
  ],
  "$elm$": undefined,
  "$flags$": 0,
  "$key$": undefined,
  "$name$": undefined,
  "$tag$": Object {},
}
`;
